k8-cluster/on-prem/rke2/ansible/README.md

```diff
@@ -0,0 +1,142 @@
+# RKE2 Kubernetes Cluster Setup using Ansible
+
+This repository contains an Ansible playbook and associated configurations to set up an RKE2 Kubernetes cluster with a primary control plane, subsequent control plane nodes, agent nodes, and an etcd cluster. The playbooks are designed to install RKE2, configure nodes, and ensure the cluster is ready for use.
+
+## Table of Contents
+- [Prerequisites](#prerequisites)
+- [Inventory Setup](#inventory-setup)
+- [Playbooks](#playbooks)
+- [Templates](#templates)
+- [Kubernetes setup via Ansible playbook](#kubernetes-setup-via-Ansible-playbook)
+- [Configuration Details](#configuration-details)
+
+## Prerequisites
+
+Before running the playbooks, ensure the following:
+
+- Install Ansible on your local machine.
+- Ensure that nodes are accessible from your local machine for SSH authentication.
```

Comment:
add playbook to check:
1. availability of ssh pem key to access all the required nodes.
2. performs the ssh connection and confirms the accesibility to the VM's.
3. presence of user with sudo priviledge in all the required instances.